runelite-client/src/main/java/net/runelite/client/plugins/gpu/GpuPlugin.java

```diff
@@ -1007,21 +1007,27 @@ public void draw()
 			{
 				if (afEnabled)
 				{
+					gl.glTexParameteri(gl.GL_TEXTURE_2D_ARRAY, gl.GL_TEXTURE_MAG_FILTER, gl.GL_LINEAR);
+					gl.glTexParameteri(gl.GL_TEXTURE_2D_ARRAY, gl.GL_TEXTURE_MIN_FILTER, gl.GL_LINEAR_MIPMAP_LINEAR);
+
 					switch (anisotropicFilteringMode)
 					{
 						case BILINEAR:
 							gl.glTexParameteri(gl.GL_TEXTURE_2D_ARRAY, gl.GL_TEXTURE_MIN_FILTER, gl.GL_LINEAR_MIPMAP_NEAREST);
 							break;
 						case TRILINEAR:
+							break;
 						default:
-							gl.glTexParameteri(gl.GL_TEXTURE_2D_ARRAY, gl.GL_TEXTURE_MIN_FILTER, gl.GL_LINEAR_MIPMAP_LINEAR);
-
-							if (anisotropicFilteringMode != anisotropicFilteringMode.TRILINEAR && gl.isExtensionAvailable("GL_EXT_texture_filter_anisotropic"))
+							if (gl.isExtensionAvailable("GL_EXT_texture_filter_anisotropic"))
 							{
 								final float maxSamples = glGetFloat(gl, gl.GL_MAX_TEXTURE_MAX_ANISOTROPY_EXT);
 								final float samples = Math.min(anisotropicFilteringMode.getSamples(), maxSamples);
 								gl.glTexParameterf(gl.GL_TEXTURE_2D_ARRAY, gl.GL_TEXTURE_MAX_ANISOTROPY_EXT, samples);
 							}
+							else
+							{
+								log.warn("Ansiotropic filtering specified but no support detected. Defaulting to trilinear filtering.");
```

Comment:
If reached, would this log on each `draw()` call? That is, 50x per second? 😬

Reply:
See line 1006 in the version you commented. The warning log only happens in this situation if the check for lastAnisotropicFilteringMode is not equal to anisotropicFilteringMode, so this warning should only happen when one changes the ansiotropic setting in Runelite and doesn't support ansiotropic filtering properly.

On a sidenote, the check is a bit pointless given practically all video cards made within the last decade has the extension and OpenGL 4.6 mandates ansiotropic support, but it's just a little bit of defensive programming for the truly esoteric GPU/driver.